### PR TITLE
added uart output

### PR DESCRIPTION
**Pull request by yverzun**

### CODE REVIEW

File: platform/omap/entry.c [hunk 1]
```diff
@@ -16,12 +16,16 @@
 #define HWADDR(addr) *((volatile uint32_t *)(addr))
 
 #include <stdint.h>
+#include <uart.h>
 
 int platform_entry(void)
 {
+    puts("initialiging...");
```
Comment (RostakaGmfun) [inline platform/omap/entry.c]: Typo

File: platform/omap/include/uart.h [hunk 2]
```diff
@@ -0,0 +1,190 @@
+#ifndef UART_H
+#define UART_H
+
+#define UART0 ((volatile char *)0x44E09000)
```
Comment (RostakaGmfun) [inline platform/omap/include/uart.h]: Can you please add UART1, UART2, etc addresses as well?

File: platform/omap/include/uart.h [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+#define UARTx_OP_R_RHR 			0x0
```
Comment (RostakaGmfun) [inline platform/omap/include/uart.h]: Please replace tabs with spaces and align all defines (hope your editor can do it automatically)

File: platform/omap/uart.c [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#include <uart.h>
+
+void puts(char *s) {
```
Comment (RostakaGmfun) [inline platform/omap/uart.c]: `const char*`

File: platform/omap/include/uart.h (hunk 2 above)
Comment (yverzun) [inline platform/omap/include/uart.h]: np

File: platform/omap/include/uart.h [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+#define UARTx_MB_W_CFPS         0x60
+#define UARTx_MB_W_RXFIFO_LVL   0x64
+#define UARTx_MB_W_TXFIFO_LVL   0x68
+#define UARTx_MB_W_IER2         0x6C
+#define UARTx_MB_W_ISR2         0x70
+#define UARTx_MB_W_FREQ_SEL     0x74
+#define UARTx_MB_W_MDR3         0x80
+#define UARTx_MB_W_TXDMA        0x84
+
+void puts(const char *s);
```
Comment (RostakaGmfun) [inline platform/omap/include/uart.h]: Please add `-fnobuiltin` compiler flag or make this function return `int` (number of bytes written).

File: platform/omap/entry.c [hunk 6]
```diff
@@ -16,12 +16,16 @@
...
+#include <uart.h>
 
 int platform_entry(void)
 {
+    puts("initializing...");
     // enable functional clock and GPIO1 module
     HWADDR(OMAP_PER_CLOCK_GPIO1) |= 2;
 
+
+
```
Comment (RostakaGmfun) [inline platform/omap/entry.c]: Remove black lines

File: platform/omap/include/clock.h [hunk 7]
```diff
@@ -0,0 +1,145 @@
+#ifndef CLOCK_H
+#define CLOCK_H
+
+#include <stdint.h>
+
+#define DISABLE_MODULE 0x00
```
Comment (RostakaGmfun) [inline platform/omap/include/clock.h]: Let's put these defines into enum too.

File: platform/omap/include/pin_mode.h [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+{
+   MODE_0 = 0,
+   MODE_1 = 1,
+   MODE_2 = 2,
+   MODE_3 = 3,
+   MODE_4 = 4,
+   MODE_5 = 5,
+   MODE_6 = 6,
+   MODE_7 = 7 
+}pinmode;
```
Comment (RostakaGmfun) [inline platform/omap/include/pin_mode.h]: Please add space after `}` and rename all types into form of `snake_case_t`

**Comment by yverzun** [comment]
closed. need to be fixed a bit.

**Pull request closed without merging**